### DISCUSSION TITLE
I need feedback with my structure folders (react)

### DISCUSSION THREAD

**Discussion by AguHDS** [community/community#147094] [Programming Help]
### Body

I'm building a personal react app and i need feedback of people that knows about react structure folders. A friend told me that my structure is unorganized and hard to understand, i tried my best to make it easy to understand.
You don't need to see the code, just the structure folder to see if it make sense, i tried to see other projects and follow good practices, but i never got feedback and i'm working by my own, so i need help with it. 

Please take a look to the folders name and tell me if it make sense for you (the app is meant to be way bigger in the future)

[
![1](https://github.com/user-attachments/assets/53b3a983-b048-43b7-a4a5-9739c003d48e)
![2](https://github.com/user-attachments/assets/c8a9e89a-ff9a-4613-a14c-78552010e82b)
![3](https://github.com/user-attachments/assets/18a6e042-c01b-4724-a9b2-e8d3e508e81e)
![4](https://github.com/user-attachments/assets/4dbc5658-63c6-4bf4-b724-baf0b4e3bdbb)
![5](https://github.com/user-attachments/assets/88fe024f-d976-44d4-8f94-0e30d283effd)
![6](https://github.com/user-attachments/assets/898e1890-b32a-4dda-a485-6f0d638afa96)
](url)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by baoanh177** [comment]
You can refer to this structure I am using
![code](https://github.com/user-attachments/assets/e6ec0152-f6c8-4225-a163-80d73bd052ad)

In `/components/common/Button/index.jsx` why I not use `Button.jsx` instead of `index.jsx`

Because when importing, if there is an index.jsx file inside, you just need to import like this `import ... from "@/components/common/Button"` instead of `import ... from "@/components/common/Button/Button.jsx"`. Is it cleaner than right?

So what about `/components/common/index.js` and other files like that. As I noted, it will import and export components that are at its same level. It will have lines of code like this `export * from "./Button"`, It is responsible for centralizing all components into one file _(To use `export * from "./Button"` syntax you need to export all components by `export` not `export default`)_ and when you need to import Button component just import like this `import { Button } from "@/components/common"`, similar to file `/components/index.js` you can import like `import { Button, DefaultLayout, AuthLayout } from "@/components"`

- Note: Use `export` instead of `export default` for the entire project, `index.jsx` instead of `Button.jsx`

**Comment by Logophoman** [comment]
Personally - I don't think you're structure is too bad it mostly comes down to personal preference and granularity of grouping things. I personally end up with like 30 components in one folder while others split it up so there's often only 3-4 in one. I think anyone willing to look at your repository will need to dig a bit anyway so it doesn't matter too much imo...

**Reply by baoanh177** [reply]
Yea, the projects in my repository don't seem to follow the structure I mentioned because I recently shared the structure with others. I've used this structure for a private project and for me it's pretty clean. So if you have any ideas or structures that you find interesting, please share them for me and everyone to refer to. That will be my pleasure.
